src/components/pages/CategoryList.js

```diff
@@ -1,25 +1,68 @@
-import React from "react";
-import { List, ListItem, ListItemText, withStyles } from "@material-ui/core/";
+import React, { useEffect, useState } from "react";
+import {
+  List,
+  ListItem,
+  ListItemText,
+  makeStyles,
+  CircularProgress,
+} from "@material-ui/core/";
 import TopAppBar from "../pageComponents/TopAppBar";
 import BottomNavigationBar from "../pageComponents/BottomNavigationBar";
 import ExpandableListItem from "../pageComponents/ExpandableListItem";
+import * as ContentProvider from "../services/contentProvider";
 
-const styles = (theme) => ({
+const useStyles = makeStyles((theme) => ({
   container: {
     marginStart: "16px",
     marginEnd: "16px",
   },
   text: {
     color: theme.palette.text.main,
   },
-});
+}));
 
-class CategoryList extends React.Component {
-  render() {
-    const { classes, title, categories, history, match } = this.props;
-    return (
-      <div>
-        <TopAppBar data-testid="appbar" title={title} />
+export default function CategoryList(props) {
+  const classes = useStyles();
+  const { category, history, match } = props;
+  const [loading, setLoading] = useState(true);
+  const [title, setTitle] = useState("");
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    let unsubscribe = () => {};
+    if (category === undefined) {
+      console.log("UNDEF");
+      unsubscribe = ContentProvider.getStructure({
+        next: (querySnapshot) => {
+          let appStructure = [];
+          let structure = querySnapshot.docs;
+          console.log(structure);
+          Promise.all(
+            structure.map((category) => appStructure.push(category.data()))
+          ).then(() => {
+            const category = appStructure.find(
+              (element) => element.value === match.params.category
+            );
+            setTitle(category.title);
+            setCategories(category.subcategories);
+            setLoading(false);
```

Comment:
I could be mistaken, but I believe this information should be present on the structure context and does not need to be fetched again. 

If I wouldn't be on a train I would test it myself ... sry for that.

Reply:
It is only fetched, if the structure is not in the context... could be if the user enters the app via a link...